content/guides/autodiscovery.md

```diff
@@ -48,155 +86,169 @@ By default, the Datadog Agent includes Autodiscovery support for:
+etcdctl set /datadog/check_configs/httpd/init_configs '[{}, {}]'
+etcdctl set /datadog/check_configs/httpd/instances '[{"apache_status_url": "http://%%host%%/server-status?auto"},{"name": "My service", "url": "http://%%host%%", timeout: 1}]'
+~~~
 
-Note that in the structure above, you may have multiple checks for a single container. For example you may run a Java service that provides an HTTP API, using the HTTP check and the JMX integration at the same time. To declare that in templates, simply add elements to the `check_names`, `init_configs`, and `instances lists`. These elements will be matched together based on their index in their respective lists.
+Again, the list orders matter. The HTTP check will only work if all its elements have the same index (1) across the lists (they do).
 
-### Example: Apache Web Server
+### Template Source: Kubernetes Pod Annotations
 
-By default, the Datadog Agent supports Autodiscovery for the Apache Web Server through the [`conf.d/auto_conf/apache.yaml` file](https://github.com/DataDog/integrations-core/blob/master/apache/conf.yaml.example):
+Since version 5.12 of the Datadog Agent, you can store check templates in Kubernetes pod annotations. Autodiscovery detects if the Agent is running on Kubernetes and searches all pod annotations for templates if so; you don't have to configure it as a template source as you do with key-value stores.
 
-    docker_images:
-      - httpd
+Follow the [Kubernetes integration instructions](/integrations/kubernetes/), then add annotations to your pod definitions. Autodiscovery expects annotations to look like the following:
 
-    init_config:
+~~~
+annotations:
+  service-discovery.datadoghq.com/<Kubernetes Container Name>.check_names: '[<CHECK_NAME>]'
+  service-discovery.datadoghq.com/<Kubernetes Container Name>.init_configs: '[<INIT_CONFIG>]'
+  service-discovery.datadoghq.com/<Kubernetes Container Name>.instances: '[<INSTANCE_CONFIG>]'
+~~~
 
-    instances:
-      - apache_status_url: http://%%host%%/server-status?auto
+It's very similar to the template format for key-value stores. Autodiscovery looks for annotation keys beginning with `service-discovery.datadoghq.com/`. Kubernetes container names function as the service identifiers.
 
-To store the same configuration template in etcd you could run the following commands:
+The following pod annotation defines two templates—equivalent to those from the end of the previous section—for `apache` containers:
 
-    etcdctl mkdir /datadog/check_configs/httpd
-    etcdctl set /datadog/check_configs/httpd/check_names '["apache"]'
-    etcdctl set /datadog/check_configs/httpd/init_configs '[{}]'
-    etcdctl set /datadog/check_configs/httpd/instances '[{"apache_status_url": "http://%%host%%/server-status?auto"}]'
+~~~
+apiVersion: v1
+kind: Pod
+metadata:
+  name: apache
+  annotations:
+    service-discovery.datadoghq.com/apache.check_names: '["apache","http_check"]'
```

Comment:
IIRC that's 5.14 only (must check the milestone). We could document that to avoid support cases with old versions